apps/angular/crud/src/app/data-access/todos.store.ts

```diff
@@ -0,0 +1,125 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { randText } from '@ngneat/falso';
+import { ComponentStore } from '@ngrx/component-store';
+import { Todo } from '../model/todo.interface';
+
+export interface TodosState {
+  todos: Todo[];
+  loading: boolean;
+  error: string | null;
+  disabledTodosIds: number[];
+  errorTodosIds: number[];
+}
+
+@Injectable()
+export class TodosStore extends ComponentStore<TodosState> {
+  constructor(private http: HttpClient) {
+    super({
+      todos: [],
+      loading: false,
+      error: null,
+      disabledTodosIds: [],
+      errorTodosIds: [],
+    });
+  }
+
+  load() {
+    this.patchState({ loading: true, error: null });
+    this.http
+      .get<Todo[]>('https://jsonplaceholder.typicode.com/todos')
+      .subscribe({
+        next: (todos) => {
+          this.patchState({ todos, loading: false });
+        },
+        error: (error) => {
+          this.patchState({ loading: false, error: JSON.stringify(error) });
+        },
+      });
+  }
```

Comment:
if you want to use the component store, don't use subcribe but wrap your code inside an effect

Reply:
changed to effects